src/mapping/RadialBasisFctSolver.h

```diff
@@ -0,0 +1,181 @@
+  template <typename RADIAL_BASIS_FUNCTION_T>
+  void computeDecomposition(RADIAL_BASIS_FUNCTION_T basisFunction, const mesh::Mesh &inputMesh, const mesh::Mesh &outputMesh, std::vector<bool> deadAxis);
+
+  /// Maps the given input data
+  Eigen::VectorXd solveConsistent(const Eigen::VectorXd &inputData) const;
+
+  /// Maps the given input data
+  Eigen::VectorXd solveConservative(const Eigen::VectorXd &inputData) const;
+
+  // Clear all stored matrices
+  void clear();
+
+  Eigen::MatrixXd _matrixA;
+
+private:
+  precice::logging::Logger _log{"mapping::RadialBasisFctSolver"};
+
+  Eigen::ColPivHouseholderQR<Eigen::MatrixXd> _qr;
+};
+
+template <typename RADIAL_BASIS_FUNCTION_T>
+static Eigen::MatrixXd buildMatrixCLU(RADIAL_BASIS_FUNCTION_T basisFunction, const mesh::Mesh &inputMesh, std::vector<bool> deadAxis);
+
+template <typename RADIAL_BASIS_FUNCTION_T>
+static Eigen::MatrixXd buildMatrixA(RADIAL_BASIS_FUNCTION_T basisFunction, const mesh::Mesh &inputMesh, const mesh::Mesh &outputMesh, std::vector<bool> deadAxis);
+
+template <typename RADIAL_BASIS_FUNCTION_T>
+void RadialBasisFctSolver::computeDecomposition(RADIAL_BASIS_FUNCTION_T basisFunction, const mesh::Mesh &inputMesh, const mesh::Mesh &outputMesh, std::vector<bool> deadAxis)
+{
+  // First, assemble the interpolation matrix
+  _qr = buildMatrixCLU(basisFunction, inputMesh, deadAxis).colPivHouseholderQr();
+
+  PRECICE_CHECK(_qr.isInvertible(),
+                "The interpolation matrix of the RBF mapping from mesh {} to mesh {} is not invertable. "
+                "This means that the mapping problem is not well-posed. "
+                "Please check if your coupling meshes are correct. Maybe you need to fix axis-aligned mapping setups "
+                "by marking perpendicular axes as dead?",
+                inputMesh.getName(), outputMesh.getName());
+
+  // Second, assemble evaluation matrix
+  _matrixA = buildMatrixA(basisFunction, inputMesh, outputMesh, deadAxis);
+}
+
+Eigen::VectorXd RadialBasisFctSolver::solveConservative(const Eigen::VectorXd &inputData) const
+{
+  // TODO: Avoid temporary allocations
+  PRECICE_ASSERT(inputData.size() == _matrixA.rows());
+  Eigen::VectorXd Au = _matrixA.transpose() * inputData;
+  PRECICE_ASSERT(Au.size() == _matrixA.cols());
+  return _qr.solve(Au);
+}
+
+Eigen::VectorXd RadialBasisFctSolver::solveConsistent(const Eigen::VectorXd &inputData) const
+{
+  PRECICE_ASSERT(inputData.size() == _matrixA.cols());
+  Eigen::VectorXd p = _qr.solve(inputData);
+  PRECICE_ASSERT(p.size() == _matrixA.cols());
+  return _matrixA * p;
+}
+
+void RadialBasisFctSolver::clear()
+{
+  _matrixA = Eigen::MatrixXd();
+  _qr      = Eigen::ColPivHouseholderQR<Eigen::MatrixXd>();
+}
+
+// ------- Non-Member Functions ---------
+
+/// Deletes all dead directions from fullVector and returns a vector of reduced dimensionality.
+static inline Eigen::VectorXd reduceVector(
```

Comment:
Is there a reason why you moved it from EigenHelperFunctions?
I could imagine that this will be used in the RBF partition of unity code as well.

Reply:
That's a preparation of #1320. I will need it for PU as well, but hopefully only in this file.